appserver/web/weld-integration/src/main/java/org/glassfish/weld/FilteringClassLoader.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2022 Eclipse Foundation and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.weld;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.Runtime.Version;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.jar.Attributes;
+import java.util.jar.Manifest;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.enumeration;
+import static java.util.stream.StreamSupport.stream;
+
+/**
+ * This classloader filters out extensions which are incompatible with current environment.
+ *
+ * @author David Matejcek
+ */
+class FilteringClassLoader extends ClassLoader {
+    private static final Logger LOG = Logger.getLogger(FilteringClassLoader.class.getName());
+    private static final String PATH_WELD_EXTENTSION = "META-INF/services/jakarta.enterprise.inject.spi.Extension";
+    private static final String PATH_MANIFEST = "META-INF/MANIFEST.MF";
+    private static final Pattern PATTERN_OSGI_EE_JAVA = Pattern.compile("\\(&\\(osgi.ee=JavaSE\\)\\(version=(.+)\\)\\)");
+    private static final Version JDK17 = Version.parse("17");
+
+    public FilteringClassLoader(ClassLoader parent) {
+        super(parent);
+        LOG.log(Level.FINEST, "Parent: {0}", parent);
+    }
+
+
+    @Override
+    public URL getResource(final String name) {
+        LOG.log(Level.FINE, "getResource(): {0}", name);
+        return getParent().getResource(name);
```

Comment:
This method doesn't need to filter the resources? Shouldn't this method gelegate to the `getResources` method to also filter resources here?

Reply:
I have fallen to this trap recently too, but these methods are not on the same level. They usually follow the scenario when you first get all resources by one method, then load one of them by the second one. Then you would do the filtering twice with no reason.
The class has just limited visibility, so we can do this as we know it is always used just this way (but I should add some test to cover that - test containers).